### PR TITLE
Add scheme to DNS service discovery docs

### PR DESCRIPTION
**Pull request by daquinoaldo**
The missing schema leads to errors like `improperly formatted alertmanager URL "alertmanager.mimir.svc.cluster.local:8080/alertmanager" (maybe the scheme is missing?)`.

- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — CHANGES_REQUESTED** [review]
Thanks for opening this PR! The issue you raised is interesting. The doc page `docs/sources/operators-guide/configure/about-dns-service-discovery.md` is a generic one about the DNS-based service discovery. In the example you edited there (memcached one) it's correct to **not** have a schema (e.g. memcached protocol is not HTTP based). However, when you use the DNS-based service discovery with the alertmanager, you need to set the scheme there.

We can do few things to clarify it:
1. In `docs/sources/operators-guide/configure/about-dns-service-discovery.md`, mention the scheme is required when referencing the config option `-ruler.alertmanager-url`
2. We can improve the CLI flag description [here](https://github.com/grafana/mimir/blob/8329c5cba240a16cdf4a8fe51a54573c05cceb61/pkg/ruler/ruler.go#L144) to clarify it there as well

**Comment by daquinoaldo** [comment]
Thank you, @pracucci. I've updated the PR.

**Review by pracucci — APPROVED** [review]
LGTM. Could you run `make reference-help doc` to update the auto-generated doc, please? @osg-grafana can you take a final look at the copy, please?

File: docs/sources/operators-guide/configure/about-dns-service-discovery.md [hunk 1]
```diff
@@ -35,4 +35,6 @@ You select a discovery mode by adding one of the following supported prefixes to
 - **`dnssrvnoa+`**<br />
   The domain name after the prefix is looked up as a SRV query, with no A/AAAA lookup made after that. For example: `dnssrvnoa+_memcached._tcp.memcached.namespace.svc.cluster.local`.
 
+The Alertmanager URL must contain also the scheme. For example: `dns+http://alermanager.local:8080`.
```
[the remark this answers is not in the document]
Comment (osg-grafana) [reply docs/sources/operators-guide/configure/about-dns-service-discovery.md:38]: Word order

**Review by osg-grafana — APPROVED** [review]
Unblocking with feedback

**Review by jhesketh — APPROVED** [review]

**Pull request merged**